src/runtime/utils/refreshHandler.ts

```diff
@@ -0,0 +1,79 @@
+import type { RefreshHandlerConfig, RefreshHandler } from '../types'
+import { useRuntimeConfig, useAuth, useAuthState } from '#imports'
+
+export const defaultRefreshHandler: RefreshHandler & {
+  config?: RefreshHandlerConfig
+  refetchIntervalTimer?: ReturnType<typeof setInterval>
+  refreshTokenIntervalTimer?: ReturnType<typeof setInterval>
+  visibilityHandler(): void
+} = {
+  // Session configuration keep this for reference
+  config: undefined,
+
+  // Refetch interval
+  refetchIntervalTimer: undefined,
+
+  // TODO: find more Generic method to start a Timer for the Refresh Token
+  // Refetch interval for local/refresh schema
+  refreshTokenIntervalTimer: undefined,
+
+  visibilityHandler () {
+    // Listen for when the page is visible, if the user switches tabs
+    // and makes our tab visible again, re-fetch the session, but only if
+    // this feature is not disabled.
+    if (this.config?.enableRefreshOnWindowFocus && document.visibilityState === 'visible') {
+      useAuth().getSession()
+    }
+  },
+
+  init (config: RefreshHandlerConfig): void {
+    this.config = config
+
+    const runtimeConfig = useRuntimeConfig().public.auth
+
+    const { data } = useAuthState()
+    const { getSession } = useAuth()
+
+    document.addEventListener('visibilitychange', () => {
+      this.visibilityHandler()
+    }, false)
+
+    const { enableRefreshPeriodically } = config
+
+    if (enableRefreshPeriodically !== false) {
+      const intervalTime =
+        enableRefreshPeriodically === true ? 1000 : enableRefreshPeriodically
+      this.refetchIntervalTimer = setInterval(() => {
+        if (data.value) {
+          getSession()
+        }
+      }, intervalTime)
+    }
+
+    if (runtimeConfig.provider.type === 'refresh') {
+      const intervalTime = runtimeConfig.provider.token.maxAgeInSeconds! * 1000
+      const { refresh, refreshToken } = useAuth()
+
+      this.refreshTokenIntervalTimer = setInterval(() => {
+        if (refreshToken.value) {
+          refresh()
+        }
+      }, intervalTime)
+    }
+  },
+
+  destroy (): void {
+    // Clear visibility handler
+    document.removeEventListener('visibilitychange', () => {
+      this.visibilityHandler()
+    }, false)
```

Comment:
not actually removing the handler here

Reply:
Done. Now using the function ref inside the object